shared/utils/logging.hpp

```diff
@@ -1,402 +1,80 @@
 #pragma once
 
-#include <android/log.h>
-#include <pthread.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <dlfcn.h>
-#include <unistd.h>
-#include <cmath> // Included to support cmath's definition of log
 #include <string_view>
-#include <algorithm>
-#include <string>
-#include <list>
-#include <mutex>
-#include "modloader/shared/modloader.hpp"
-#include "utils-functions.h"
-#include <thread>
-#include <unordered_set>
-#include <unordered_map>
+#include <fmt/format.h>
+#include <fmt/ranges.h>
 
-namespace Logging {
-    enum Level {
-        CRITICAL = ANDROID_LOG_FATAL,
-        ERROR = ANDROID_LOG_ERROR,
-        WARNING = ANDROID_LOG_WARN,
-        INFO = ANDROID_LOG_INFO,
-        DEBUG = ANDROID_LOG_DEBUG
-    };
-}
+#include "paper/shared/logger.hpp"
+#include "paper/shared/modinfo_fmt.hpp"
+
+#include "utils-functions.h"
 
-#ifdef log
-#undef log
-#endif
+#include "il2cpp-api-types.h"
+#include "il2cpp-class-internals.h"
 
-#ifndef LOG_PATH
-#define LOG_PATH "/sdcard/Android/data/%s/files/logs/"
-#endif
+namespace il2cpp_utils {
+    Paper::LoggerContext const& getLogger();
+    std::string ClassStandardName(const Il2CppClass* klass, bool generics);
+    const char* TypeGetSimpleName(const Il2CppType* type);
+}
 
-class Logger;
+template<>
+struct fmt::formatter<Il2CppClass const*> : formatter<std::string> {
 
-/// @class Logger Buffer
-/// @brief A buffer for logger data. Used for logging to file in a buffered fashion.
-/// Each LoggerBuffer exists to wrap around a single logger instance.
-/// Every time log is called on the instance, this buffer is updated (assuming options.toFile is true for the instance)
-/// Each buffer contains multiple messages that need to be written out, stored as an std::list
-class LoggerBuffer {
-    friend Logger;
-    public:
-    std::list<std::string> messages;
-    std::mutex messageLock;
-    const ModInfo modInfo;
-    bool closed = false;
-    static std::string get_logDir();
-    std::string get_path() {
-        std::string cpy = modInfo.version;
-        std::replace(cpy.begin(), cpy.end(), '.', '_');
-        auto val = get_logDir() + modInfo.id + "_" + cpy + ".log";
-        return val;
+    // Formats the point p using the parsed format specification (presentation)
+    // stored in this formatter.
+    template<typename FormatContext>
+    auto format(Il2CppClass const* p, FormatContext &ctx) -> decltype(ctx.out()) {
+        // ctx.out() is an output iterator to write to.
+        return formatter<std::string>::format(::il2cpp_utils::ClassStandardName(p, true), ctx);
```

Comment:
The class pointer itself is actually somewhat useful when logging the class itself. That said, that can be done rather explicitly instead, so no need to change this unless you feel like it would be helpful